Glossary.md

```diff
@@ -522,6 +522,50 @@ Another approach, manual memory management, would allow you to produce the faste
 Reference counting implementation:
 - Old compiler: [Mono folder](crates/compiler/mono/src) (search ref)
 
+## Borrow
+
+A **borrowing** function reads its argument without affecting its reference count.
+The caller retains ownership and can continue using the value after the call.
+
+Example builtins that borrow: `strEqual`, `listLen`, `strContains`
+
+See [src/builtins/OWNERSHIP.md](src/builtins/OWNERSHIP.md) for detailed ownership semantics.
+
+## Consume
+
+A **consuming** function takes ownership of its argument. The caller transfers
+ownership to the callee and must not use the argument after the call. The function
+is responsible for cleanup (decref when done).
```

Comment:
I'm not sure this is the best wording. Specifically
> The caller transfers ownership to the callee and must not use the argument after the call.

If the caller needs to use the value after calling the function, they need to increment the arguments refcount.

```ruby
x = "abc......long string"
y = x.concat(".first-ending")
z = x.concat(".second-ending")
```

`strConcat` is a consuming function in both cases, yet we are using `x` after the call.
This is valid. The interpreter is required to insert a refcount increment before the call to `strConcat` that creates `y`.